### PR TITLE
 ctl: pd-ctl cmd will work well in char device mode

### LINKED ISSUE #4478: tools: pd-ctl cmd  will work well in crontab
## Bug Report
#!/bin/sh ./.tiup/components/ctl/v5.0.0-rc/pd-ctl -u http://xxx.xx.xx.xx:2379 scheduler show is ok.

But it will work wrong if I run the cmd in crontab:

image

### What did you do?
```
 * * * * * /data2/robert/soft/golang/pd/bin/pd-ctl -u xxx:xxx:xxx:xxx:2379 scheduler show >> /data2/robert/cron.log 2>&1
```

### What did you expect to see?
It will run well in crontab 
### What version of PD are you using (`pd-server -V`)?
5.0.0~master: fe6fab92

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?
close #4478

### What is changed and how it works?
String split mention If s does not contain sep and sep is not empty, Split returns a
slice of length 1 whose only element is s. It will add args if the arg is nil in char device mode.
So it needs to check the args is nil. 
### Check List

Tests

- Manual test (add detailed scripts or steps below)

image

Code changes

Side effects

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
pdctl args lens is wrong in char device mode.
```

### CODE REVIEW

**Review by lhy1024 — COMMENTED** [review]
Do we need to add a test for it？

**Comment by bufferflies** [comment]
> Do we need to add a test for it？

I think it should need.

**Review by lhy1024 — APPROVED** [review]

File: tools/pd-ctl/pdctl/ctl.go [hunk 1]
```diff
@@ -198,3 +198,15 @@ func genCompleter(cmd *cobra.Command) []readline.PrefixCompleterInterface {
...
 	return pc
 }
+
+// ReadStdin convert stdin to string array
+func ReadStdin(r io.Reader) (input []string, err error) {
+	b, err := io.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	if s := strings.TrimSpace(string(b)); s != "" {
```
Comment (JmPotato) [inline tools/pd-ctl/pdctl/ctl.go:208]:
```suggestion
	if s := strings.TrimSpace(string(b)); len(s) > 0 {
```

File: tools/pd-ctl/main.go [hunk 2]
```diff
@@ -49,16 +47,14 @@ func main() {
...
 
-	var input []string
+	var inputs []string
 	stat, _ := os.Stdin.Stat()
 	if (stat.Mode() & os.ModeCharDevice) == 0 {
-		b, err := io.ReadAll(os.Stdin)
+		in, err := pdctl.ReadStdin(os.Stdin)
 		if err != nil {
 			fmt.Println(err)
-			return
```
Comment (JmPotato) [inline tools/pd-ctl/main.go:58]: Why not `return err` here?

File: tools/pd-ctl/pdctl/ctl.go [hunk 3]
```diff
@@ -198,3 +198,15 @@ func genCompleter(cmd *cobra.Command) []readline.PrefixCompleterInterface {
 	}
 	return pc
 }
+
+// ReadStdin convert stdin to string array
+func ReadStdin(r io.Reader) (input []string, err error) {
```
Comment (JmPotato) [inline tools/pd-ctl/pdctl/ctl.go:203]: Since the name of it is `ReadStdin`, I think there is no need to pass the `os.Stdin` through a parameter?

**Review by JmPotato — COMMENTED** [review]
The rest LGTM.

File: tools/pd-ctl/pdctl/ctl.go (hunk 3 above)
Comment (bufferflies) [reply tools/pd-ctl/pdctl/ctl.go:203]: the parameter design to test simple.

**Review by JmPotato — APPROVED** [review]

**Pull request merged**